src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/slick/SubmissionComponent.scala

```diff
@@ -69,7 +74,203 @@ trait SubmissionComponent {
     def value = foreignKey("FK_SUB_VALIDATION_VAL", valueId, attributeQuery)(_.id.?)
   }
 
-  protected val submissionQuery = TableQuery[SubmissionTable]
   protected val submissionValidationQuery = TableQuery[SubmissionValidationTable]
 
+  protected object submissionQuery extends TableQuery(new SubmissionTable(_)) {
+
+    type SubmissionQueryType = Query[SubmissionTable, SubmissionRecord, Seq]
+
+    /*
+      the core methods
+     */
+
+    /* gets a submission */
+    def get(workspaceContext: SlickWorkspaceContext, submissionId: String): ReadAction[Option[Submission]] = {
+      loadSubmission(UUID.fromString(submissionId))
+    }
+
+    /* lists all submissions in a workspace */
+    def list(workspaceContext: SlickWorkspaceContext): ReadAction[Seq[Submission]] = {
+      findSubmissionsByWorkspaceId(workspaceContext.workspaceId).result.flatMap(recs => DBIO.sequence(recs.map{ rec =>
+        loadSubmission(rec.id) map(sub => sub.get)
+      }))
+    }
+
+    /* saves a submission and associated workflows to a workspace */
+    def save(workspaceContext: SlickWorkspaceContext, submission: Submission): ReadWriteAction[Submission] = {
+
+      def saveSubmissionWorkflows(workflows: Seq[Workflow]) = {
+        DBIO.seq(workflows.map { case (wf) =>
+          workflowQuery.save(workspaceContext, submission.submissionId, wf)
+        }.toSeq: _*)
+      }
+
+      uniqueResult[SubmissionRecord](findSubmissionById(UUID.fromString(submission.submissionId))) flatMap {
+        case None =>
+          val configId = uniqueResult(methodConfigurationQuery.findConfigByName(
+            workspaceContext.workspaceId, submission.methodConfigurationNamespace, submission.methodConfigurationName).map(_.id))
+
+          loadSubmissionEntityId(workspaceContext.workspaceId, submission.submissionEntity) flatMap { eId =>
+            configId flatMap { cId =>
+              (submissionQuery += marshalSubmission(workspaceContext.workspaceId, submission, eId, cId.get))
+            } andThen
+            saveSubmissionWorkflows(submission.workflows)
+          }
+        case Some(submissionRec) =>
+          update(workspaceContext, submission)
```

Comment:
save calls update, update calls save... seems bad. Do we need an update? Should we just call submissionQuery.update here and remove the update function?